src/MySqlConnector/MySqlDataReader.cs

```diff
@@ -604,6 +606,9 @@ internal async Task DisposeAsync(IOBehavior ioBehavior, CancellationToken cancel
 			Command.CancellableCommand.SetTimeout(Constants.InfiniteTimeout);
 			connection.FinishQuerying(m_hasWarnings);
 
+			m_activity.SetSuccess();
```

Comment:
Here's another interesting thing: [specification for database providers](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md) does not mention whether statement is completed whenever the first row is received, or every single row is read. SqlClient does the former.

Reply:
Yes, I noted that here: https://github.com/mysql-net/MySqlConnector/issues/1036:

> TODO: Does the `Execute` activity end when `ExecuteReader` returns a `DbDataReader`, or when that `DbDataReader` has finished consuming the result set (and is closed/disposed)?

My preference is the latter; as an application developer, I usually care about when the entire `using (var reader = cmd.ExecuteReader()) { ... }` block has finished, not about when it "begins".

I suggested adding an event for when the `DbDataReader` is created and returned, for those who care about breaking the whole span into two parts.

It would be ideal to standardise this, have a good name for the event, etc.

Reply:
See ongoing conversation in https://github.com/npgsql/npgsql/pull/3967#discussion_r705255298